app/livechat/server/hooks/leadCapture.ts

```diff
@@ -1,8 +1,22 @@
 import { callbacks } from '../../../../lib/callbacks';
-import { settings } from '../../../settings';
-import { LivechatVisitors } from '../../../models';
+import { settings } from '../../../settings/server/index';
+import { LivechatVisitors } from '../../../models/server/index';
 
-function validateMessage(message, room) {
+type MessageData = {
+	editedAt: Date;
+	token: string;
+	t: string;
+	msg: string;
+};
+type RoomData = {
+	t: string;
+	v: {
+		token: string;
+		_id: string;
+	};
+};
+
+function validateMessage(message: MessageData, room: RoomData): boolean {
```

Comment:
Try to use existing types. Here `message` should be `IMessage` and room `IRoom`. If you use IRoom and you find a prop that may not exists on IRoom but exists on IOmnichannelRoom, you can use a "type guard" like the ones we have on IRoom.ts


```
function validateMessage(message: IMessage, room: IRoom): boolean {
   // using room.v (as v only exists on Omnichannel rooms) will throw an error. to avoid that, we use the type guard
	if (!isOmnichannelRoom(room)) { return message // do nothing if room is not omnichannel room }
 	const x = room.v // room.v will work normally :)
```

Reply:
Alright, it worked! Implemented in d49bf2abb7451df60b635c0fcc8f2c85ee745617